pkg/build/builder/common_test.go

```diff
@@ -197,6 +197,19 @@ func Test_addBuildParameters(t *testing.T) {
 			original: `# no FROM instruction`,
 			want:     want{},
 		},
+		{
+			original: heredoc.Doc(`
+				ARG GOLANG_CONTAINER=golang:latest
+				FROM $GOLANG_CONTAINER
+				RUN echo "hello world"
+				`),
+			want: want{
+				Out: heredoc.Doc(`
+				FROM $GOLANG_CONTAINER
```

Comment:
Note the actual substitution does not happen as part of the method we call below.

See https://github.com/openshift/builder/blob/master/vendor/github.com/openshift/imagebuilder/dispatchers.go#L519-L523

Reply:
Actually something may be amiss here.  See below.